main/commands.js

```diff
@@ -105,6 +105,7 @@ async function history(message, args) {
     }
 
     let filter = target == "global" ? m => true : m => m.sender == target || m.recipient == target
+    filter.slice(-10);
```

Comment:
`slice` returns a copy of the array, it doesn't mutate the array itself
So you need to do `filter = filter.slice(-10);`